File: backend/utils/token_utils.py
from datetime import datetime, timedelta, timezone
from typing import Optional
import jwt
import os


SECRET_KEY = os.environ.get('JWT_SECRET_KEY', 'your-secret-key-change-in-production')
ALGORITHM = 'HS256'
ACCESS_TOKEN_EXPIRE_MINUTES = 60 * 24 * 7


def create_access_token(data: dict, expires_delta: Optional[timedelta] = None) -> str:
    """Create a JWT access token"""
    to_encode = data.copy()
    
    if expires_delta:
        expire = datetime.now(timezone.utc) + expires_delta
    else:
        expire = datetime.now(timezone.utc) + timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
    
    to_encode.update({"exp": expire})
    encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
    return encoded_jwt


def decode_access_token(token: str) -> Optional[dict]:
    """Decode and validate a JWT access token"""
    try:
        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
        return payload
    except jwt.ExpiredSignatureError:
        return None
    except jwt.JWTError:
        return None

File: backend/src/utils/token.utils.ts
import jwt from 'jsonwebtoken';
import { JWTPayload } from '../interfaces/user.interface';

const SECRET_KEY = process.env.JWT_SECRET || 'your-secret-key-change-in-production';
const EXPIRES_IN = process.env.JWT_EXPIRES_IN || '7d';

export class TokenUtils {
  static createAccessToken(payload: JWTPayload): string {
    return jwt.sign(payload, SECRET_KEY, { expiresIn: EXPIRES_IN });
  }

  static verifyAccessToken(token: string): JWTPayload | null {
    try {
      const payload = jwt.verify(token, SECRET_KEY) as JWTPayload;
      return payload;
    } catch (error) {
      return null;
    }
  }
}
